source/solvers/cahn_hilliard_assemblers.cc

```diff
@@ -5,135 +5,86 @@
 #include <solvers/copy_data.h>
 
 
+
 template <int dim>
 void
 CahnHilliardAssemblerCore<dim>::assemble_matrix(
   CahnHilliardScratchData<dim> &scratch_data,
   StabilizedMethodsCopyData    &copy_data)
 {
   // Gather physical properties
-  const double well_height       = this->cahn_hilliard_parameters.well_height;
-  const double mobility_constant = this->mobility_constant;
-  const double epsilon           = scratch_data.epsilon;
+  const double epsilon   = this->epsilon;
+  const double cell_size = scratch_data.cell_size;
+  const double xi =
+    this->cahn_hilliard_parameters.potential_smoothing_coefficient;
```

Comment:
Why is epsilon passed to the constructor explicitely but the smoothign parameter is still within the parameters? This should be explained a bit more above.

Reply:
The smoothing parameter is always specified by the user, so it is kept in the parameters. Epsilon on the contrary can either be user specified or automatically determined. For the latter, the minimum cell diameter must be known : it is accessible through the minimum_cell_diameter function (GridTools class) used on the triangulation object, which, itself, is an attribute of the CahnHilliard class, but the the assembler.

Reply:
Should I clarify this somewhere in the code ?

Reply:
I would add a comment or something similar to make it specific why this is done this way